sdk-go/wflib/wf_lib_internal.go

```diff
@@ -558,6 +558,94 @@ func (n *NodeOutput) jsonPathImpl(path string) NodeOutput {
 	}
 }
 
+func (n *NodeOutput) handleExceptionOnChild(handler ThreadFunc, exceptionName *string) {
+	n.thread.checkIfIsActive()
+	node := n.thread.spec.Nodes[n.nodeName]
+	if node.GetWaitForThreads() == nil {
+		n.thread.throwError(errors.New("can only call handleExceptionOnChild on WaitForThreads Node"))
+	}
+
+	var threadName string
+	if exceptionName != nil {
+		threadName = "exn-handler-" + n.nodeName + "-" + *exceptionName
+	} else {
+		threadName = "exn-handler-" + n.nodeName
+	}
+	threadName = n.thread.wf.addSubThread(threadName, handler)
```

Comment:
This is a perfect case to follow the rule of three, a refactor might be a good thing to do since all three follow the same pattern.

Reply:
Tried it...it's actually pretty complex to get it right for all three cases. I think it's more readable this way.